aredis/pool.py

```diff
@@ -350,11 +358,17 @@ def get_connection(self, command_name, *keys, **options):
 
         return connection
 
-    def make_connection(self, node):
+    async def make_connection(self, node):
         """
         Create a new connection
         """
-        if self.count_all_num_connections(node) >= self.max_connections:
+        wait_for_connection = self.wait_for_connection
+        while self.count_all_num_connections(node) >= self.max_connections:
+            if wait_for_connection > 0:
+                wait_for_connection -= 1
+                await asyncio.sleep(1)
```

Comment:
Personally, I don't think so -- pretty much everything is just plumbing at that point; we don't really lose anything by making it all async and it makes it easier to make other things async in the future, if need be.

That said, maybe its too much? This could also be a `time.sleep(1)` instead, but that would cause a lot of blocking, so wouldn't really be the best solution...